### PR TITLE
Refactoring naming in workflows

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Update description of workflow input
- Rename fetch-tracks to fetch-repos
- Rename tracks input to repos
- Remove unused environment variable
- Rename track matrix value by repo
- Rename TRACK_REPO_FULLNAME to TARGET_REPO_FULLNAME
- Rename track_repo to target_repo

### CODE REVIEW

File: .github/workflows/sync-tooling.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   workflow_dispatch:
     inputs:
       tracks:
-        description: 'List of repos to rerun the syncing workflow for, formatted as JSON array, e.g. ["exercism/julia"]'
+        description: 'List of tooling repos to rerun the syncing workflow for, formatted as JSON array, e.g. ["exercism/julia"]'
```
Comment (SaschaMann) [inline .github/workflows/sync-tooling.yml:7]: `workflow_dispatch` will take whatever repos you give it, it doesn't apply a filter.

File: .github/workflows/sync-tracks.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ on:
   workflow_dispatch:
     inputs:
       tracks:
-        description: 'List of repos to rerun the syncing workflow for, formatted as JSON array, e.g. ["exercism/julia"]'
+        description: 'List of track repos to rerun the syncing workflow for, formatted as JSON array, e.g. ["exercism/julia"]'
```
Comment (SaschaMann) [inline .github/workflows/sync-tracks.yml:7]: `workflow_dispatch` will take whatever repos you give it, it doesn't apply a filter.

File: .github/workflows/sync-rest.yml [hunk 3]
```diff
@@ -71,7 +71,7 @@ jobs:
             // TODO: The GraphQL API does have this info. Look into changing to that.
             const botForks = botRepos.flatMap(({ full_name, fork }) => fork ? [full_name.replace(process.env.BOT_USERNAME, 'exercism')] : [])
 
-            const tracks = ${{ steps.fetch-tracks.outputs.result }}
+            const tracks = ${{ steps.fetch-repos.outputs.result }}
```
Comment (SaschaMann) [inline .github/workflows/sync-rest.yml:74]: Should the JS variables also be renamed?

File: .github/workflows/sync-tooling.yml [hunk 4]
```diff
@@ -89,16 +89,16 @@ jobs:
...
   abort-timer:
-    needs: fetch-tracks
+    needs: fetch-repos
     runs-on: ubuntu-20.04
     name: ⚠ 5 MINUTES TO ABORT
     environment: abort-timer
     steps:
       - run: exit 0
 
   push-to-tracks:
```
Comment (SaschaMann) [inline .github/workflows/sync-tooling.yml:99]: This still mentions tracks.

**Review by SaschaMann — COMMENTED** [review]
There are a few places that still mention track, but I think you know and this is just a first step?

Other than the description, this looks good to me. I reviewed it commit-by-commit so I couldn't add suggestions to fix the description.

**Review by SaschaMann — APPROVED** [review]

File: .github/workflows/sync-rest.yml (hunk 3 above)
Comment (ErikSchierboom) [reply .github/workflows/sync-rest.yml]: I've renamed

**Comment by ErikSchierboom** [comment]
> There are a few places that still mention track, but I think you know and this is just a first step?

Correct. I want to get these relatively simple renames to be done first.

**Review by SaschaMann — APPROVED** [review]
LGTM

**Pull request merged**